diff --git a/android/build.gradle b/android/build.gradle
index bed09d62d..78cc876d7 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -12,7 +12,10 @@ buildscript {
 }

 apply plugin: 'com.android.library'
-apply plugin: 'org.jetbrains.kotlin.android'
+def builtInKotlinEnabled = !project.hasProperty("android.builtInKotlin") || project.property("android.builtInKotlin").toString().toBoolean()
+if (!builtInKotlinEnabled) {
+  apply plugin: 'org.jetbrains.kotlin.android'
+}

 def isNewArchitectureEnabled() {
   // To opt-in for the New Architecture, you can either:
@@ -38,7 +41,7 @@ android {
   if (agpVersion.tokenize('.')[0].toInteger() >= 7) {
     namespace "com.reactnativegooglesignin"
   }
-  compileSdkVersion getExtOrIntegerDefault('compileSdkVersion')
+  compileSdk getExtOrIntegerDefault('compileSdkVersion')

   // Used to override the NDK path/version on internal CI or by allowing
   // users to customize the NDK path/version from their root project (e.g. for M1 support)
@@ -50,7 +53,7 @@ android {
   }

   defaultConfig {
-    minSdkVersion getExtOrIntegerDefault('minSdkVersion')
-    targetSdkVersion getExtOrIntegerDefault('targetSdkVersion')
+    minSdk getExtOrIntegerDefault('minSdkVersion')
+    targetSdk getExtOrIntegerDefault('targetSdkVersion')
     buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString()
   }
