diff --git a/android/build.gradle b/android/build.gradle
index 89232d7d070103d1e32a67d55cf394dfbfddb060..5498c9219d0f3d794b63a6bcb3d2a88a97540070 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -2,11 +2,9 @@ apply plugin: 'com.android.library'
 apply plugin: 'expo-module-gradle-plugin'
 apply plugin: "expo-autolinking"
 
-buildscript {
-  // Simple helper that allows the root project to override versions declared by this library.
-  ext.safeExtGet = { prop, fallback ->
-    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
-  }
+// Simple helper that allows the root project to override versions declared by this library.
+project.ext.safeExtGet = { prop, fallback ->
+  rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
 }
 
 group = 'host.exp.exponent'
@@ -19,7 +17,10 @@ expoModule {
 
 android {
   namespace "expo.core"
+  compileSdk = project.ext.safeExtGet("compileSdkVersion", 36)
   defaultConfig {
+    minSdk = project.ext.safeExtGet("minSdkVersion", 24)
+    targetSdk = project.ext.safeExtGet("targetSdkVersion", 36)
     versionCode 1
     versionName "55.0.9"
     consumerProguardFiles("proguard-rules.pro")
diff --git a/ios/AppDelegates/EXReactRootViewFactory.h b/ios/AppDelegates/EXReactRootViewFactory.h
index 2daec5d4629ba0db42b2fff9da28d78e5f71cf7c..df3fdfe4a83f23a4c6f7d408975d78654c27bb03 100644
--- a/ios/AppDelegates/EXReactRootViewFactory.h
+++ b/ios/AppDelegates/EXReactRootViewFactory.h
@@ -28,6 +28,7 @@ NS_SWIFT_NAME(ExpoReactRootViewFactory)
 - (UIView *)superViewWithModuleName:(NSString *)moduleName
                   initialProperties:(nullable NSDictionary *)initialProperties
                       launchOptions:(nullable NSDictionary *)launchOptions
+                bundleConfiguration:(RCTBundleConfiguration *)bundleConfiguration
                devMenuConfiguration:(nullable RCTDevMenuConfiguration *)devMenuConfiguration;
 #else
 - (UIView *)superViewWithModuleName:(NSString *)moduleName
diff --git a/ios/AppDelegates/EXReactRootViewFactory.mm b/ios/AppDelegates/EXReactRootViewFactory.mm
index 6fecc8df0a98a85c3a28bfa740af2eeab7ba11b9..3d97c595b991230df0d0deaf11f0699a242755ba 100644
--- a/ios/AppDelegates/EXReactRootViewFactory.mm
+++ b/ios/AppDelegates/EXReactRootViewFactory.mm
@@ -35,24 +35,34 @@ - (instancetype)initWithReactDelegate:(nullable EXReactDelegate *)reactDelegate
 - (UIView *)viewWithModuleName:(NSString *)moduleName
              initialProperties:(nullable NSDictionary *)initialProperties
                  launchOptions:(nullable NSDictionary *)launchOptions
+           bundleConfiguration:(RCTBundleConfiguration *)bundleConfiguration
           devMenuConfiguration:(RCTDevMenuConfiguration *)devMenuConfiguration
 {
   if (self.reactDelegate != nil) {
     return [self.reactDelegate createReactRootViewWithModuleName:moduleName initialProperties:initialProperties launchOptions:launchOptions];
   }
-  return [super viewWithModuleName:moduleName initialProperties:initialProperties launchOptions:launchOptions devMenuConfiguration:devMenuConfiguration];
+  return [super viewWithModuleName:moduleName
+                 initialProperties:initialProperties
+                     launchOptions:launchOptions
+               bundleConfiguration:bundleConfiguration
+              devMenuConfiguration:devMenuConfiguration];
 }
 
 - (UIView *)superViewWithModuleName:(NSString *)moduleName
                   initialProperties:(nullable NSDictionary *)initialProperties
                       launchOptions:(nullable NSDictionary *)launchOptions
+                bundleConfiguration:(RCTBundleConfiguration *)bundleConfiguration
                devMenuConfiguration:(nullable RCTDevMenuConfiguration *)devMenuConfiguration
 {
   if (devMenuConfiguration == nil) {
     devMenuConfiguration = [RCTDevMenuConfiguration defaultConfiguration];
   }
 
-  return [super viewWithModuleName:moduleName initialProperties:initialProperties launchOptions:launchOptions devMenuConfiguration:devMenuConfiguration];
+  return [super viewWithModuleName:moduleName
+                 initialProperties:initialProperties
+                     launchOptions:launchOptions
+               bundleConfiguration:bundleConfiguration
+              devMenuConfiguration:devMenuConfiguration];
 }
 #else
 - (UIView *)viewWithModuleName:(NSString *)moduleName
diff --git a/ios/AppDelegates/ExpoReactNativeFactory.swift b/ios/AppDelegates/ExpoReactNativeFactory.swift
index 925e0f9a44cf160543b76967a094b2cc3221da7c..11f56595c40de37cb87f3e4d75c22aab29a7a969 100644
--- a/ios/AppDelegates/ExpoReactNativeFactory.swift
+++ b/ios/AppDelegates/ExpoReactNativeFactory.swift
@@ -112,13 +112,15 @@ public class ExpoReactNativeFactory: ExpoReactNativeFactoryObjC, ExpoReactNative
     if let factory = self.rootViewFactory as? ExpoReactRootViewFactory {
       // RCTDevMenuConfiguration is only available in react-native 0.83+
 #if os(iOS) || os(tvOS)
+      let devMenuConfiguration = self.devMenuConfiguration ?? RCTDevMenuConfiguration.default()
       // When calling `recreateRootViewWithBundleURL:` from `EXReactRootViewFactory`,
       // we don't want to loop the ReactDelegate again. Otherwise, it will be an infinite loop.
       rootView = factory.superView(
         withModuleName: moduleName ?? defaultModuleName,
         initialProperties: initialProps,
         launchOptions: launchOptions ?? [:],
-        devMenuConfiguration: self.devMenuConfiguration
+        bundleConfiguration: self.bundleConfiguration,
+        devMenuConfiguration: devMenuConfiguration
       )
 #else
       rootView = factory.superView(
@@ -133,7 +135,8 @@ public class ExpoReactNativeFactory: ExpoReactNativeFactoryObjC, ExpoReactNative
         withModuleName: moduleName ?? defaultModuleName,
         initialProperties: initialProps,
         launchOptions: launchOptions,
-        devMenuConfiguration: self.devMenuConfiguration
+        bundleConfiguration: self.bundleConfiguration,
+        devMenuConfiguration: self.devMenuConfiguration ?? RCTDevMenuConfiguration.default()
       )
 #else
       rootView = rootViewFactory.view(
