diff --git a/android/build.gradle b/android/build.gradle
index f226765..39ca858 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -12,7 +12,7 @@ buildscript {
 }
 
 apply plugin: 'com.android.library'
-apply plugin: 'kotlin-android'
+apply plugin: 'org.jetbrains.kotlin.android'
 
 def getExtOrDefault(name) {
     return rootProject.ext.has(name) ? rootProject.ext.get(name) : project.properties['Purchases_' + name]
@@ -23,11 +23,11 @@ def getExtOrIntegerDefault(name) {
 }
 
 android {
-    compileSdkVersion getExtOrIntegerDefault('compileSdkVersion')
+    compileSdk getExtOrIntegerDefault('compileSdkVersion')
 
     defaultConfig {
-        minSdkVersion getExtOrIntegerDefault('minSdkVersion')
-        targetSdkVersion getExtOrIntegerDefault('targetSdkVersion')
+        minSdk getExtOrIntegerDefault('minSdkVersion')
+        targetSdk getExtOrIntegerDefault('targetSdkVersion')
         versionCode 1
         versionName '9.15.0'
     }
@@ -37,7 +37,7 @@ android {
             minifyEnabled false
         }
     }
-    lintOptions {
+    lint {
         disable 'GradleCompatible'
     }
     compileOptions {
